### PR TITLE
Add EC2 userdata test workflow

### PR DESCRIPTION
**Pull request by zhihonl**
# Description of the issue
In order to migrate internal base integration tests onto GitHub action, a separate workflow needs to be created to test EC2 user data for CWAgent since terraform has to create a separate instance that has the user data attached to it.

# Description of changes
This adds a workflow for terraform to generate an instance with the userdata that installs CWAgent on launch. 

# License
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

# Tests
This workflow has been tested on a [fork action run](https://github.com/zhihonl/amazon-cloudwatch-agent/actions/runs/4941700970/jobs/8835906449) and passed the metric_value_benchmark, cloudwatchlogs and run_as_user tests.
**cloudwatchlogs**
```
null_resource.integration_test (remote-exec): --- PASS: TestWriteLogsToCloudWatch (82.81s)
null_resource.integration_test (remote-exec):     --- PASS: TestWriteLogsToCloudWatch/Happy_path (41.10s)
null_resource.integration_test (remote-exec):     --- PASS: TestWriteLogsToCloudWatch/Client-side_log_filtering (41.67s)
null_resource.integration_test (remote-exec): --- PASS: TestRotatingLogsDoesNotSkipLines (71.67s)
null_resource.integration_test (remote-exec): PASS
```

**metric_value_benchmark**
```
null_resource.integration_test (remote-exec): --- PASS: TestMetricValueBenchmarkSuite (1396.15s)
null_resource.integration_test (remote-exec):     --- PASS: TestMetricValueBenchmarkSuite/TestAllInSuite (1396.15s)
null_resource.integration_test (remote-exec): PASS
```

**run_as_user**
```
null_resource.integration_test (remote-exec): --- PASS: TestRunAsUser (47.42s)
null_resource.integration_test (remote-exec):     --- PASS: TestRunAsUser/resource_file_location_resources/default.json_user_root (15.83s)
null_resource.integration_test (remote-exec):     --- PASS: TestRunAsUser/resource_file_location_resources/root.json_user_root (15.78s)
null_resource.integration_test (remote-exec):     --- PASS: TestRunAsUser/resource_file_location_resources/cwagent.json_user_cwagent (15.81s)
null_resource.integration_test (remote-exec): PASS
```
# Requirements
_Before commit the code, please do the following steps._
1. Run `make fmt` and `make fmt-sh`
2. Run `make linter`

### CODE REVIEW

File: .github/workflows/integration-test.yml [hunk 1]
```diff
@@ -551,7 +554,7 @@ jobs:
 
   EC2LinuxIntegrationTest:
     needs: [MakeBinary, StartLocalStack, GenerateTestMatrix]
-    name: 'Test'
```
Comment (sethAmazon) [inline .github/workflows/integration-test.yml:554]: This so on the actions ui we can see more of the test info

Comment (zhihonl) [reply .github/workflows/integration-test.yml:554]: This is renamed as "EC2LinuxIntegrationTest" for easier understanding.

File: .github/workflows/integration-test.yml [hunk 2]
```diff
@@ -624,6 +627,81 @@ jobs:
           retry_wait_seconds: 5
           command: cd terraform/ec2/linux && terraform destroy --auto-approve
 
+  EC2UserDataIntegrationTest:
```
[the remark this answers is not in the document]
Comment (zhihonl) [reply .github/workflows/integration-test.yml:635]: I agree. A lot of jobs we have in this workflow are just copy/paste of each other.

File: .github/workflows/integration-test.yml [hunk 3]
```diff
@@ -629,6 +632,82 @@ jobs:
...
+            else
+              terraform destroy -auto-approve && exit 1
+            fi
+      #This is here just in case workflow cancel
+      - name: Terraform destroy
+        if: ${{ cancelled() || failure() }}
+        uses: nick-fields/retry@v2
+        with:
+          max_attempts: 3
+          timeout_minutes: 8
```
Comment (SaxyPandaBear) [inline .github/workflows/integration-test.yml:707]: I'm fine with this, but do we know that these tests run in < 8 minutes? Just don't want to update this often cause it's just noise

**Review by SaxyPandaBear — APPROVED** [review]

File: .github/workflows/integration-test.yml (hunk 3 above)
Comment (zhihonl) [reply .github/workflows/integration-test.yml:707]: It should run in < 1 minute based on this [test run](https://github.com/zhihonl/amazon-cloudwatch-agent/actions/runs/5026544164/jobs/9015214589)

**Review by sky333999 — APPROVED** [review]

**Pull request merged**